src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,159 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserRegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+    private String expectedMessage;
+    private User newUser;
+
+    @BeforeEach
+    void init() {
+        newUser = new User();
```

Comment:
This way you do not need to initialize all the fields by hands in each test, and thus you can just set a single field in the test itself
```suggestion
        validUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
```